app/controllers/shelters_controller.rb

```diff
@@ -12,28 +12,29 @@ def new
   end
 
   def create
-    if Shelter.create(shelter_params).valid?
+     shelter = Shelter.create(shelter_params)
+    if shelter.valid?
       flash[:notice] = "You've successfully created a shelter."
       redirect_to '/shelters'
-    else   
-      flash[:notice] = "The following fields are missing : #{Shelter.empty_params(shelter_params)}"
+    else
+      flash[:error] = "#{shelter.errors.full_messages.to_sentence}"
```

Comment:
when called on an object that was not correctly instantiated/updated `errors.full_messages.to_sentence` returns a sentence indicating which fields were missing to create/update the object